
### Workflow GTDB-Tk

gtdbtk classify_wf --genome_dir 01_FASTA/REFORMATED --out_dir 03_TAXONOMY/GTDB-Tk -x fa --cpus 56 --skip_ani_screen 

# Add extension arguement -x and number of threads --cpus
# --skip_ani_screen allow to skip comparison genome ANI with Mash database, a time-consuming stage and unecessary for us


### Creation of an alignment file of the MAGs selected for the phylogenomic tree

gtdbtk identify --cpus 33 -x fa --genome_dir 04_MAGs/MAGs_SELECTED --out_dir 03_TAXONOMY/GTDB-Tk/02_selected/identify_bis 

gtdbtk align --cpus 33 --identify_dir 03_TAXONOMY/GTDB-Tk/02_selected/identify --out_dir 03_TAXONOMY/GTDB-Tk/02_selected/align_bis 






